src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -109,6 +109,11 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
     setMessage(undefined);
     try {
       const newTransaction = await fn(payload);
+      // TODO: OKTA-538791 this is a temp work around until the auth-js fix
+      if (!newTransaction.nextStep && newTransaction.availableSteps?.length) {
+        // eslint-disable-next-line prefer-destructuring
+        newTransaction.nextStep = newTransaction.availableSteps[0];
```

Comment:
Is this trying to get you to write it this way?
```suggestion
        [newTransaction.nextStep] = newTransaction.availableSteps;
```

Reply:
ah yeah thats what it was looking for thank you!